server.go

```diff
@@ -336,6 +341,54 @@ func (s *Server) EnforceRetentionPolicies() {
 	}
 }
 
+// StartShardGroupsPreCreate launches shard group pre-create to avoid write bottlenecks.
+func (s *Server) StartShardGroupsPreCreate(checkInterval time.Duration) error {
+	if checkInterval == 0 {
+		return fmt.Errorf("shard group pre-create check interval must be non-zero")
+	}
+	sgpcDone := make(chan struct{}, 0)
+	s.sgpcDone = sgpcDone
+	go func() {
+		for {
+			select {
+			case <-sgpcDone:
+				return
+			case <-time.After(checkInterval):
+				s.ShardGroupPreCreate(checkInterval)
+			}
+		}
+	}()
+	return nil
+}
+
+// ShardGroupPreCreate ensures that future shard groups and shards are created and ready for writing
+// is removed from the server.
+func (s *Server) ShardGroupPreCreate(checkInterval time.Duration) {
```

Comment:
This was after Phillip pointed it out though.  He fixed his on a PR between my PR so I didn't see it the first time through.  Good catch!